src/sentry/replays/usecases/query/conditions/event_ids.py

```diff
@@ -0,0 +1,169 @@
+from __future__ import annotations
+
+from uuid import UUID
+
+from snuba_sdk import Column, Condition, Function, Op
+
+from sentry.replays.lib.new_query.utils import (
+    contains,
+    does_not_contain,
+    to_uuid,
+    translate_condition_to_function,
+)
+from sentry.replays.usecases.query.conditions.base import ComputedBase
+from sentry.replays.usecases.query.conditions.error_ids import ErrorIdsArray, has_error_id
+
+
+class InfoIdScalar(ComputedBase):
+    """Look at both debug_id and info_id if info_id is queried"""
+
+    event_id_columns: list[str] = ["info_id", "debug_id"]
+
+    @classmethod
+    def visit_eq(cls, value: UUID) -> Condition:
+        return Condition(
+            Function(
+                "or",
+                _make_conditions_from_column_names(cls.event_id_columns, Op.EQ, to_uuid(value)),
+            ),
+            Op.EQ,
+            1,
+        )
```

Comment:
You might consider returning the merged function rather than the condition.  This would give you greater flexibility for re-use further down.  You could wrap with a simple helper function like `is_true_condition(merged_function)` which would just be `Condition(arg, Op.EQ, 1)`.

Reply:
looking at this i'm not sure that it simplifies it for me -- i feel even more confused looking at this helper method compared to the code that I originally wrote. mind if i follow up with this?

Reply:
No need to follow up.  You don't have to adopt that approach.  👍